packages/admin-ui/src/components/Filter/components/FilterCheckbox/useFilterCheckbox.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useCallback, useMemo, useRef, useEffect } from 'react'
+import type { UseFilterStateReturn, FilterItem } from '../Filter'
+import { usePopoverState } from '../Popover'
+import { Item } from '@react-stately/collections'
+
+import { useListBox } from '@react-aria/listbox'
+import type { ListState } from '@react-stately/list'
+import { useListState } from '@react-stately/list'
+
+export function useFilterCheckbox<T extends FilterItem>(
```

Comment:
I would change the name to `useMultipleFilter`, to disassociate from how the component renders it, to focus on the behavior.